libbeat/beat/beat.go

```diff
@@ -287,55 +283,43 @@ func (bc *instance) cleanup() error {
 }
 
 // launch manages the lifecycle of the beat and guarantees the order in which
-// the Beater methods are invokes and ensures a a proper exit code is set when
-// an error occurs. The exit flag controls if this method calls os.Exit when
-// it completes.
-func (bc *instance) launch(exit bool) error {
-	var err error
-	if exit {
-		defer func() { exitProcess(err) }()
-	}
+// the Beater methods are invoked. If an error occurs in the lifecycle of the
+// Beat it will be returned.
+func (bc *instance) launch() (err error) {
+	defer func() { err = handleError(err) }()
```

Comment:
As far as I understand this, the err passed to handleError has the value of any error happening below. And as the return param is named, err returned by handleError is the error that is returned in the end. AWESOME :-)